storage/series/series.go

```diff
@@ -399,9 +399,11 @@ func (s *dbSeries) mergeBlock(
 		return
 	}
 
-	// We are performing this in a lock, cannot wait for the existing
-	// block potentially to be retrieved from disk, lazily merge the stream.
-	newBlock.Merge(existingBlock)
+	if err := newBlock.Merge(existingBlock); err != nil {
```

Comment:
Since there's no one to float the error back to we kind of have to do this here.  FWIW this is what we were doing before when merge could return an error instead of lazily be evaluated.